### PR TITLE
Added API for creation new custom wepons

### PR DESCRIPTION
**Pull request by AnriiAndrushko**

### CODE REVIEW

File: modweaver.api/API/NewWeaponsManager.cs [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+                }
+            }
+            [HarmonyPostfix]
+            public static void Postfix(ref WeaponSpawner __instance) {
+                foreach (var w in _newWeapons) {
+                    w.gameObject.SetActive(false);
+                }
+            }
+        }
+        public enum WeaponType {
```
Comment (Ecorous) [inline modweaver.api/API/NewWeaponsManager.cs:149]: Do we really need to define our own enum?

**Comment by AnriiAndrushko** [comment]
fixed bug, now must work properly

File: modweaver.api/API/NewWeaponsManager.cs [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using TMPro;
+using Unity.Netcode;
+using UnityEngine.UI;
+using UnityEngine;
+
+namespace modweaver.api.API {
+    public static class NewWeaponsManager {
+        private static uint _prefabHandlerId = 7777; //TODO: fix possible collision
```
Comment (Ecorous) [inline modweaver.api/API/NewWeaponsManager.cs:12]: I'm not sure how we'll fix this- but I think it's fine for now. If we run into any issues then this'll need to be changed.

File: modweaver.api/API/WeaponHandler.cs [hunk 3]
```diff
@@ -0,0 +1,27 @@
+﻿using HarmonyLib;
+using Unity.Netcode;
+using UnityEngine;
+
+//dont forget: id must be same on client and server
+internal class WeaponNetworkPrefabInstanceHandler : INetworkPrefabInstanceHandler {
```
Comment (Ecorous) [inline modweaver.api/API/WeaponHandler.cs]: I'd suggest moving internal classes out of the API folder

File: modweaver.testmod/ModMain.cs [hunk 4]
```diff
@@ -1,51 +1,65 @@
...
 namespace modweaver.testmod {
     [ModMainClass]
     // ReSharper disable once UnusedType.Global (because it is used!!)
     public class ModMain : Mod {
+
+        public static List<global::Weapon> newWeapons;
+        CustomWeapon weapon1;
+
         public override void Init() {
-            //test logger
```
Comment (Ecorous) [inline modweaver.testmod/ModMain.cs:12]: Why did you delete the logging tests? It's important to test all aspects of the api in the testmod.

File: modweaver.testmod/ModMain.cs [hunk 5]
```diff
@@ -1,51 +1,65 @@
...
-
-            var test1 = Config.get("test1", -1);
-            var test2 = Config.get("test2", false, "otherfile");
-            
-            Logger.Debug("Config test values: {V1}, {V2}", test1, test2);
+            Logger.Info("Testmod is ready!");
         }
 
         public override void OnGUI(ModsMenuPopup ui) {
-            ui.CreateTitle("I'm a cool mod :3");
```
Comment (Ecorous) [inline modweaver.testmod/ModMain.cs:39]: Same with this, why delete stuff that isn't related?

File: modweaver.testmod/ModMain.cs [hunk 6]
```diff
@@ -1,51 +1,65 @@
...
-        
-        [HarmonyPostfix]
-        public static void Postfix(ref VersionNumberTextMesh __instance) {
-            __instance.textMesh.text += "\nhello world :3";
+        [HarmonyPatch(typeof(VersionNumberTextMesh), "Start")]
+        public static class VNTMPatch {
+
+            [HarmonyPostfix]
+            public static void Postfix(ref VersionNumberTextMesh __instance) {
+                __instance.textMesh.text += $"\n<color=red> Testmod v0.0.1 by YourName</color>";
```
Comment (Ecorous) [inline modweaver.testmod/ModMain.cs]: Also with this, changing things for no real reason

**Review by Ecorous — CHANGES_REQUESTED** [review]
Please don't change things that aren't related to your PR. I've marked those out where I can, however I did miss a few. I won't merge this until those are fixed.

File: modweaver.api/API/NewWeaponsManager.cs (hunk 1 above)
Comment (AnriiAndrushko) [reply modweaver.api/API/NewWeaponsManager.cs:149]: There is already enum in Weapon.WeaponType but it only describe general type and my enum describe all existing weapons and corresponds to their index in array

**Comment by AnriiAndrushko** [comment]
now must be fine, but when i tryed to test it i figure out that harmony patches not calling from API lib so that code that patch weapons into game never executed, i actually dont know how to fix this

**Comment by AnriiAndrushko** [comment]
i think it will be good to create separate "indev" branch and fix it there

**Comment by AnriiAndrushko** [comment]
Moving this PR into API repo

**Pull request closed without merging**